staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -595,7 +595,10 @@ func (r *Request) WatchWithSpecificDecoders(wrapperDecoderFn func(io.ReadCloser)
 		return nil, fmt.Errorf("for request '%+v', got status: %v", url, resp.StatusCode)
 	}
 	wrapperDecoder := wrapperDecoderFn(resp.Body)
-	return watch.NewStreamWatcher(restclientwatch.NewDecoder(wrapperDecoder, embeddedDecoder)), nil
+	return watch.NewStreamWatcher(
+		restclientwatch.NewDecoder(wrapperDecoder, embeddedDecoder),
+		errors.NewClientErrorReporter(http.StatusInternalServerError, r.verb),
```

Comment:
Why is it a 500 error?
IIUC, the error code is added at the client-side, so I don't know if there is a better error code, but it's worth documenting why we pick 500.

Reply:
Ok

Reply:
Updated, and I also return a unique status cause along with UnexpectedServerResponse - ClientWatchDecoding so a client can key on it in the future.

Reply:
PTAL